cdc/processor/pipeline/table_actor.go

```diff
@@ -0,0 +1,436 @@
+	targetTs      model.Ts
+
+	started bool
+	stopped uint32
+
+	changefeedVars   *cdcContext.ChangefeedVars
+	globalVars       *cdcContext.GlobalVars
+	replConfig       *serverConfig.ReplicaConfig
+	tableActorRouter *actor.Router
+
+	pullerNode *pullerNode
+	sortNode   *sorterNode
+	sinkNode   *sinkNode
+
+	nodes    []*ActorNode
+	actorID  actor.ID
+	stopFunc func(err error)
+
+	lastFlushTime time.Time
+	lck           sync.Mutex
+}
+
+// NewTableActor creates a table actor.
+func NewTableActor(cdcCtx cdcContext.Context,
+	mounter entry.Mounter,
+	tableID model.TableID,
+	tableName string,
+	replicaInfo *model.TableReplicaInfo,
+	sink sink.Sink,
+	targetTs model.Ts,
+) (TablePipeline, error) {
+	config := cdcCtx.ChangefeedVars().Info.Config
+	cyclicEnabled := config.Cyclic != nil && config.Cyclic.IsEnabled()
+	info := cdcCtx.ChangefeedVars()
+	vars := cdcCtx.GlobalVars()
+
+	actorID := vars.TableActorSystem.ActorID()
+	mb := actor.NewMailbox(actorID, defaultOutputChannelSize)
+	// Cancel should be able to release all sub-goroutines in this actor.
+	ctx, cancel := context.WithCancel(cdcCtx)
+	// All sub-goroutines should be spawn in this wait group.
+	wg, cctx := errgroup.WithContext(ctx)
+	table := &tableActor{
+		// all errors in table actor will be reported to processor
+		reportErr: cdcCtx.Throw,
+		mb:        mb,
+		wg:        wg,
+		cancel:    cancel,
+
+		tableID:       tableID,
+		markTableID:   replicaInfo.MarkTableID,
+		tableName:     tableName,
+		cyclicEnabled: cyclicEnabled,
+		memoryQuota:   serverConfig.GetGlobalServerConfig().PerTableMemoryQuota,
+		mounter:       mounter,
+		replicaInfo:   replicaInfo,
+		replConfig:    config,
+		sink:          sink,
+		targetTs:      targetTs,
+		started:       false,
+
+		changefeedVars:   info,
+		globalVars:       vars,
+		tableActorRouter: vars.TableActorSystem.Router(),
+		actorID:          actorID,
+	}
+	table.stopFunc = stop(cctx, table)
+
+	startTime := time.Now()
+	log.Info("spawn and start table actor",
+		zap.String("changfeed", info.ID),
+		zap.String("tableName", tableName),
+		zap.Int64("tableID", tableID))
+	if err := table.start(cctx); err != nil {
+		table.stopFunc(err)
+		return nil, errors.Trace(err)
+	}
+	err := vars.TableActorSystem.System().Spawn(mb, table)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	log.Info("spawn and start table actor done",
+		zap.String("changfeed", info.ID),
+		zap.String("tableName", tableName),
+		zap.Int64("tableID", tableID),
+		zap.Duration("duration", time.Since(startTime)))
+	return table, nil
+}
+
+func (t *tableActor) Poll(ctx context.Context, msgs []message.Message) bool {
+MSG:
+	for i := range msgs {
+		if atomic.LoadUint32(&t.stopped) == stopped {
+			// No need to handle remaining messages.
+			break
+		}
+
+		switch msgs[i].Tp {
+		case message.TypeBarrier:
+			t.sortNode.UpdateBarrierTs(msgs[i].BarrierTs)
+			if err := t.sinkNode.UpdateBarrierTs(ctx, msgs[i].BarrierTs); err != nil {
+				t.checkError(err)
+				break
+			}
```

Comment:
Could you extract methods for each message? Eg.

```go
func (t *tableActor) handleBarrierMsg(msg) { }
```

Reply:
done